src/App.js

```diff
@@ -11,39 +13,263 @@ class App extends React.Component {
       cardDeck: makeShuffledDeck(),
       // currCards holds the cards from the current round
       currCards: [],
+      player1Score: 0,
+      player2Score: 0,
+      currWinner: null,
+      hasGameBegin: false,
+      numMatchesPlayer1Won: 0,
+      numMatchesPlayer2Won: 0,
+      matchNumber: 1,
     };
   }
 
   dealCards = () => {
     // this.state.cardDeck.pop() modifies this.state.cardDeck array
-    const newCurrCards = [this.state.cardDeck.pop(), this.state.cardDeck.pop()];
+    let { cardDeck } = this.state;
+    const newCurrCards = [cardDeck.pop(), cardDeck.pop()];
+    this.setState(
+      {
+        currCards: newCurrCards,
+        hasGameBegin: true,
+      },
+      () => this.calculateScoresPerRound()
+    );
+  };
+
+  calculateScoresPerRound = () => {
+    let [p1Card, p2Card] = this.state.currCards;
+    let { cardDeck } = this.state;
+    if (cardDeck.length > 0) {
+      if (p1Card.rank > p2Card.rank) {
+        this.setState((prevState) => ({
+          currWinner: 1,
+          player1Score: prevState.player1Score + 1,
+        }));
+      } else if (p1Card.rank < p2Card.rank) {
+        this.setState((prevState) => ({
+          currWinner: 2,
+          player2Score: prevState.player2Score + 1,
+        }));
+      }
+    } else if (cardDeck.length === 0) {
+      this.setState(
+        {
+          hasGameBegin: false,
+        },
+        () => this.determineFinalWinner()
+      );
+    }
```

Comment:
I think something for the future would be to rewrite if/else spaghetti whenever we can. I like to return early if possible. Here maybe a suggestion, though bit tricky here.
```suggestion
   if (!card.length) {
       return this.setState(
        {
          hasGameBegin: false,
        },
        () => this.determineFinalWinner()
      );
   }
   
   if (p1Card.rank > p2Card.rank) {
        this.setState((prevState) => ({
          currWinner: 1,
          player1Score: prevState.player1Score + 1,
        }));
      } else if (p1Card.rank < p2Card.rank) {
        this.setState((prevState) => ({
          currWinner: 2,
          player2Score: prevState.player2Score + 1,
        }));
      }
```

In cases where we wouldn't have to worry about the tie situation, we probably could simplify this even further.